I confirm the error mentioned above.
```
  sync_actions.go:97:         INFO [04-14|12:13:18.081] no blocks in state                       role=batcher syncStatus.headL1=1fff81..0a7c28:5 syncStatus.currentL1=1fff81..0a7c28:5 syncStatus.localSafeL2=000000..000000:0 syncStatus.safeL2=34bbc8..16724f:0 syncStatus.unsafeL2=2445b5..86af44:10 syncActions="SyncActions{blocksToPrune: 0, channelsToPrune: 0, clearState: nil, blocksToLoad: [1, 10]}"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12db558]
```

--- a/espresso/environment/espresso_docker_helpers.go
+++ b/espresso/environment/espresso_docker_helpers.go
@@ -9,9 +9,14 @@ import (
 	"fmt"
 	"io"
 	"os/exec"
+	"runtime"
 	"strings"
 	"time"
 )
+
+// This is a reliable way to determine if we are running on Linux as a runtime
+// check.
+var isRunningOnLinux = runtime.GOOS == "linux"
 
 // DockerContainerInfo is a struct that contains information about a Docker
 // Container that was launched by the DockerCli struct.
@@ -37,7 +42,8 @@ type DockerContainerConfig struct {
 
 	Ports []string
 
-	AutoRM bool
+	Network string
+	AutoRM  bool
 }
 
 // DockerCli is a simple implementation of a Docker Client that is used to
@@ -67,8 +73,14 @@ func (d *DockerCli) LaunchContainer(ctx context.Context, config DockerContainerC
 			args = append(args, "--rm")
 		}
 
-		for _, port := range config.Ports {
-			args = append(args, "-p", port)
+		if config.Network != "" {
+			args = append(args, "--network", config.Network)
+		}
+
+		if config.Network != "host" {
+			for _, port := range config.Ports {
+				args = append(args, "-p", port)
+			}
 		}
 
 		for key, value := range config.Environment {
@@ -117,7 +129,17 @@ func (d *DockerCli) LaunchContainer(ctx context.Context, config DockerContainerC
 
 	portMap := map[string][]string{}
 	containerInfo := DockerContainerInfo{ContainerID: containerID, PortMap: portMap}
-	{
+	if config.Network == "host" {
+		// If we're running on the host network, we don't need to do anything
+		// special to get the ports.  They are the same as the ones we specified
+		// in the config.
+
+		for _, port := range config.Ports {
+			portMap[port] = []string{
+				fmt.Sprintf("0.0.0.0:%s", port),
+			}
+		}
+	} else {
 		for _, portToFind := range config.Ports {
 			outputBuffer.Reset()
 			// Let's find out what our assigned ports ended up being

--- a/espresso/environment/optitmism_espresso_test_helpers.go
+++ b/espresso/environment/optitmism_espresso_test_helpers.go
@@ -10,6 +10,7 @@ import (
 	"net"
 	"net/http"
 	"net/url"
+	"strconv"
 	"testing"
 	"time"
 
@@ -412,6 +413,21 @@ func fundEspressoAccount() DevNetLauncherOption {
 	}
 }
 
+// This code is adapted from a gist file:
+// https://gist.github.com/sevkin/96bdae9274465b2d09191384f86ef39d
+func determineFreePort() (port int, err error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		err = listener.Close()
+	}()
+
+	addr := listener.Addr().(*net.TCPAddr)
+	return addr.Port, nil
+}
+
 // launchEspressoDevNodeDocker is DevNetLauncherOption that launches th
 // Espresso Dev Node within a Docker container.  It also ensures that the
 // Espresso Dev Node is actively producing blocks before returning.
@@ -453,8 +469,26 @@ func launchEspressoDevNodeDocker() DevNetLauncherOption {
 								l1EthRpcURL.Host = net.JoinHostPort("host.docker.internal", port)
 							}
 
-							l1EthRpcURL.Scheme = "http"
+							portRemapping := map[string]string{
+								ESPRESSO_BUILDER_PORT:       ESPRESSO_BUILDER_PORT,
+								ESPRESSO_SEQUENCER_API_PORT: ESPRESSO_SEQUENCER_API_PORT,
+								ESPRESSO_DEV_NODE_PORT:      ESPRESSO_DEV_NODE_PORT,
+							}
 
+							if isRunningOnLinux {
+								for portKey := range portRemapping {
+									// We need to determine a free port on the host system
+									// to bind the espresso-dev-node to.
+									freePort, err := determineFreePort()
+									if err != nil {
+										ct.Error = FailedToDetermineL1RPCURL{Cause: err}
+										return
+									}
+									portRemapping[portKey] = strconv.FormatInt(int64(freePort), 10)
+								}
+							}
+
+							l1EthRpcURL.Scheme = "http"
 							containerCli := new(DockerCli)
 
 							dockerConfig := DockerContainerConfig{
@@ -467,14 +501,14 @@ func launchEspressoDevNodeDocker() DevNetLauncherOption {
 									"ESPRESSO_SEQUENCER_STORAGE_PATH":             "/data/espresso",
 									"RUST_LOG":                                    "info",
 
-									"ESPRESSO_BUILDER_PORT":       ESPRESSO_BUILDER_PORT,
-									"ESPRESSO_SEQUENCER_API_PORT": ESPRESSO_SEQUENCER_API_PORT,
-									"ESPRESSO_DEV_NODE_PORT":      ESPRESSO_DEV_NODE_PORT,
+									"ESPRESSO_BUILDER_PORT":       portRemapping[ESPRESSO_BUILDER_PORT],
+									"ESPRESSO_SEQUENCER_API_PORT": portRemapping[ESPRESSO_SEQUENCER_API_PORT],
+									"ESPRESSO_DEV_NODE_PORT":      portRemapping[ESPRESSO_DEV_NODE_PORT],
 								},
 								Ports: []string{
-									ESPRESSO_BUILDER_PORT,
-									ESPRESSO_SEQUENCER_API_PORT,
-									ESPRESSO_DEV_NODE_PORT,
+									portRemapping[ESPRESSO_BUILDER_PORT],
+									portRemapping[ESPRESSO_SEQUENCER_API_PORT],
+									portRemapping[ESPRESSO_DEV_NODE_PORT],
 								},
 							}
 
@@ -497,6 +531,19 @@ func launchEspressoDevNodeDocker() DevNetLauncherOption {
 							}
 
 							ct.EspressoDevNode = EspressoDevNodeDockerContainerInfo(espressoDevNodeContainerInfo)
+
+							if isRunningOnLinux {
+								for portKey, portValue := range portRemapping {
+									// We copy the port mapping information
+									// so we know the original mapping again,
+									// since we're hard-coding the ports to use.
+									// This should allow us to run multiple
+									// e2e test environments in parallel on
+									// linux as well.
+									espressoDevNodeContainerInfo.PortMap[portKey] = espressoDevNodeContainerInfo.PortMap[portValue]
+
+								}
+							}
 
 							// We have all of our ports.
 							// Let's return all of the relevant port mapping information

--- a/espresso/environment/optitmism_espresso_test_helpers.go
+++ b/espresso/environment/optitmism_espresso_test_helpers.go
@@ -28,7 +28,8 @@ import (
 // "const ESPRESSO_DEV_NODE_DOCKER_IMAGE = "ghcr.io/espressosystems/espresso-sequencer/espresso-dev-node:release-newfoundland"
 // "const ESPRESSO_DEV_NODE_DOCKER_IMAGE = "ghcr.io/espressosystems/espresso-sequencer/espresso-dev-node:release-labrador"
 // const ESPRESSO_DEV_NODE_DOCKER_IMAGE = "ghcr.io/espressosystems/espresso-sequencer/espresso-dev-node:release-builder"
-const ESPRESSO_DEV_NODE_DOCKER_IMAGE = "ghcr.io/espressosystems/espresso-sequencer/espresso-dev-node:20241115"
+// const ESPRESSO_DEV_NODE_DOCKER_IMAGE = "ghcr.io/espressosystems/espresso-sequencer/espresso-dev-node:20241115"
+const ESPRESSO_DEV_NODE_DOCKER_IMAGE = "ghcr.io/espressosystems/espresso-sequencer/espresso-dev-node:release-goldendoodle"
 
 // deployed ESPRESSO_SEQUENCER_LIGHT_CLIENT_ADDRESS at 0x17435cce3d1b4fa2e5f8a08ed921d57c6762a180
 // deployed ESPRESSO_SEQUENCER_PLONK_VERIFIER_ADDRESS at 0xb4b46bdaa835f8e4b4d8e208b6559cd267851051
@@ -336,7 +337,7 @@ func allowHostDockerInternalVirtualHost() DevNetLauncherOption {
 						// We append the host machine address to the list of virtual hosts, so
 						// that we do not get denied when attempting to access the host machine's
 						// RPC API.
-						nodeCfg.HTTPVirtualHosts = append(nodeCfg.HTTPVirtualHosts, "host.docker.internal")
+						nodeCfg.HTTPVirtualHosts = append(nodeCfg.HTTPVirtualHosts, "host.docker.internal", "localhost")
 
 						return nil
 					},
@@ -445,12 +446,18 @@ func launchEspressoDevNodeDocker() DevNetLauncherOption {
 
 							// We replace the host with host.docker.internal to inform
 							// docker to communicate with the host system.
-							l1EthRpcURL.Host = net.JoinHostPort("host.docker.internal", port)
+
+							if isRunningOnLinux {
+								l1EthRpcURL.Host = net.JoinHostPort("localhost", port)
+							} else {
+								l1EthRpcURL.Host = net.JoinHostPort("host.docker.internal", port)
+							}
+
 							l1EthRpcURL.Scheme = "http"
 
 							containerCli := new(DockerCli)
 
-							espressoDevNodeContainerInfo, err := containerCli.LaunchContainer(ct.Ctx, DockerContainerConfig{
+							dockerConfig := DockerContainerConfig{
 								Image: ESPRESSO_DEV_NODE_DOCKER_IMAGE,
 								Environment: map[string]string{
 									"ESPRESSO_DEPLOYER_ACCOUNT_INDEX":             ESPRESSO_MNEMONIC_INDEX,
@@ -469,7 +476,20 @@ func launchEspressoDevNodeDocker() DevNetLauncherOption {
 									ESPRESSO_SEQUENCER_API_PORT,
 									ESPRESSO_DEV_NODE_PORT,
 								},
-							})
+							}
+
+							if isRunningOnLinux {
+								// We launch in network mode host on linux,
+								// otherwise the container is not able to
+								// communicate with the host system.
+								// We use host.docker.internal to do this on
+								// platforms that are not running natively on
+								// linux, as this special address achieves the
+								// same result.  But on linux, this does not
+								// work, and we need to run on the host instead.
+								dockerConfig.Network = "host"
+							}
+							espressoDevNodeContainerInfo, err := containerCli.LaunchContainer(ct.Ctx, dockerConfig)
 
 							if err != nil {
 								ct.Error = FailedToLaunchDockerContainer{Cause: err}
@@ -512,7 +532,7 @@ func launchEspressoDevNodeDocker() DevNetLauncherOption {
 							currentBlockHeightURLString := "http://" + hostPort + "/status/block-height"
 
 							// Wait for Espresso to be ready
-							timeoutCtx, cancel := context.WithTimeout(ct.Ctx, time.Minute*10)
+							timeoutCtx, cancel := context.WithTimeout(ct.Ctx, 3*time.Minute)
 							defer cancel()
 							if err := WaitForEspressoBlockHeightToBePositive(timeoutCtx, currentBlockHeightURLString); err != nil {
 								ct.Error = EspressoNodeFailedToBecomeReady{Cause: err}

